tests/integration/test_sqs.py

```diff
@@ -2020,6 +2061,96 @@ def test_fifo_content_based_message_deduplication_arrives_once(
         )
         assert response.get("Messages", []) == []
 
+    @pytest.mark.aws_validated
+    @pytest.mark.parametrize("content_based_deduplication", [True, False])
+    def test_fifo_deduplication_arrives_once_after_delete(
+        self,
+        sqs_create_queue,
+        aws_client,
+        content_based_deduplication,
+        snapshot,
+    ):
+        attributes = {"FifoQueue": "true"}
+        if content_based_deduplication:
+            attributes["ContentBasedDeduplication"] = "true"
+        queue_url = sqs_create_queue(
+            QueueName=f"test-queue-{short_uid()}.fifo",
+            Attributes=attributes,
+        )
+        item = '{"foo": "bar"}'
+        group = "group-1"
+        kwargs = {}
+        if not content_based_deduplication:
+            kwargs["MessageDeduplicationId"] = "dedup1"
+        aws_client.sqs.send_message(
+            QueueUrl=queue_url, MessageBody=item, MessageGroupId=group, **kwargs
+        )
+
+        # first receive has the item
+        response = aws_client.sqs.receive_message(QueueUrl=queue_url, WaitTimeSeconds=3)
+        snapshot.match("get-messages", response)
+        assert len(response["Messages"]) == 1
+        assert response["Messages"][0]["Body"] == item
+        # delete the item, we want to check that we don't receive a duplicate even after deletion
+        # see https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SendMessage.html
+        aws_client.sqs.delete_message(
+            QueueUrl=queue_url, ReceiptHandle=response["Messages"][0]["ReceiptHandle"]
+        )
+
+        # republish the same message
+        aws_client.sqs.send_message(
+            QueueUrl=queue_url, MessageBody=item, MessageGroupId=group, **kwargs
+        )
+
+        # second doesn't receive anything since the message has the same content
```

Comment:
nit: since we parametrize content_based_deduplication, I'd say that the "second doesn't receive anything because the deduplication id is the same" (which is derived from the content in on of the cases, but not in the other)